packages/sharing-smart-contract/test/e2e/consumeProtectedData.test.js

```diff
@@ -137,47 +137,138 @@ describe('ConsumeProtectedData', () => {
+    // TODO: when _spender has not approved the contract => should revert the consumeProtectedData tx
+    it('should revert if the consumer has not approved the DataProtectorSharingContract', async () => {
+      const { dataProtectorSharingContract, protectedDataAddress, appAddress, collectionTokenId, subscriptionParams } =
+        await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({ dataProtectorSharingAddress });
+
+      await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .subscribeToCollection(collectionTokenId, subscriptionParams);
 
-        await dataProtectorSharingContract
+      expect(
+        dataProtectorSharingContract
           .connect(voucherOwner)
-          .subscribeToCollection(collectionTokenId, subscriptionParams);
+          .consumeProtectedData(protectedDataAddress, workerpoolOrder, appAddress, true),
+      ).to.be.reverted;
+    });
+  });
+
+  describe('voucher - consumeProtectedData()', () => {
+    it('should create a deal on chain and consume asset in the user voucher', async () => {
+      const {
+        dataProtectorSharingContract,
+        pocoContract,
+        protectedDataAddress,
+        appAddress,
+        collectionTokenId,
+        subscriptionParams,
+      } = await loadFixture(createCollectionWithProtectedDataRatableAndSubscribable);
+      const dataProtectorSharingAddress = await dataProtectorSharingContract.getAddress();
+      const { voucherOwner, workerpoolOrder } = await createVoucher({ dataProtectorSharingAddress });
+
+      await pocoContract
+        .connect(voucherOwner)
+        .approve(await dataProtectorSharingContract.getAddress(), subscriptionParams.price);
+      await pocoContract.connect(voucherOwner).deposit({
+        value: ethers.parseUnits(subscriptionParams.price.toString(), 'gwei'),
+      }); // value sent should be in wei
+
+      await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .subscribeToCollection(collectionTokenId, subscriptionParams);
+
+      const tx = await dataProtectorSharingContract
+        .connect(voucherOwner)
+        .consumeProtectedData(protectedDataAddress, workerpoolOrder, appAddress, true);
+      await tx.wait();
+
+      expect(tx)
+        .to.emit(dataProtectorSharingContract, 'ProtectedDataConsumed')
+        .withArgs((_dealId, _protectedDataAddress, _mode) => {
+          assert.equal(_dealId.constructor, ethers.Bytes32, 'DealId should be of type bytes32');
+          assert.equal(_protectedDataAddress, protectedDataAddress, 'DealId should be of type bytes32');
+          assert.equal(_mode, 0, 'Mode should be SUBSCRIPTION (0)');
+        });
+    });
+
+    it('should revert if the consumer has approved contract but account balance is insufficient', async () => {
```

Comment:
in the case of the voucher, the voucher owner must `authorize` DataProtectorSharing to use its voucher balance

the title should be
```suggestion
    it('should revert if the consumer has authorized the contract to use its voucher but the balance is insufficient', async () => {
```